### PR TITLE
grass: 8.2.0 -> 8.2.1 and fix build failure

### PR DESCRIPTION
**Pull request by nh2**
###### Description of changes

Fixes https://github.com/NixOS/nixpkgs/pull/215960#issuecomment-1568854089

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/gis/grass/default.nix [hunk 1]
```diff
@@ -6,24 +6,33 @@
...
   pname = "grass";
-  version = "8.2.0";
+  version = "8.2.1";
 
   src = with lib; fetchFromGitHub {
     owner = "OSGeo";
     repo = "grass";
     rev = version;
-    sha256 = "sha256-VK9FCqIwHGmeJe5lk12lpAGcsC1aPRBiI+XjACXjDd4=";
+    sha256 = "sha256:02fxcpdwkjis0i5mqks1by4kwazi1947flmhfd0yz2xxgdvx0wsk";
```
Comment (imincik) [inline pkgs/applications/gis/grass/default.nix:15]: I suggest changing `src.sha256` attribute to `src.hash` which I believe is the newer standard.

Also, the hash value I've got by using `lib.fakeHash` trick doesn't match hash in this PR:
```
    hash = "sha256-U3PQd3u9i+9Bc7BSd0gK8Ss+iV9BT1xLBDrKydtl3Qk=";
```

File: pkgs/applications/gis/grass/default.nix [hunk 2]
```diff
@@ -6,24 +6,33 @@
...
 
   nativeBuildInputs = [
     pkg-config bison flex makeWrapper wrapGAppsHook
-    gdal geos libmysqlclient netcdf pdal
+    gdal # for `gdal-config`
+    geos # for `geos-config`
+    netcdf # for `nc-config`
+    libmysqlclient # for `mysql_config`
+    pdal # for `pdal-config`; remove with next version, see https://github.com/NixOS/nixpkgs/pull/215960#issuecomment-1568854089
   ] ++ (with python3Packages; [ python-dateutil numpy wxPython_4_2 ]);
```
Comment (imincik) [inline pkgs/applications/gis/grass/default.nix:25]: Changes in `nativeBuildInputs` look ok, the only thing I suggest to change is comment on `pdal` line. I suggest to point it directly to the [GRASS PR](https://github.com/OSGeo/grass/pull/2851) where they made change from `pdal-config` to `pkg-config`.

File: pkgs/applications/gis/grass/default.nix [hunk 3]
```diff
@@ -6,24 +6,33 @@
...
   buildInputs = [
     cairo zlib proj libtiff libpng fftw sqlite
     readline ffmpeg postgresql blas wxGTK32
     proj-datumgrid zstd
+    gdal
+    geos
+    netcdf
+    libmysqlclient
+    pdal
   ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
```
Comment (imincik) [inline pkgs/applications/gis/grass/default.nix:36]: Looks good to me.

**Review by imincik — CHANGES_REQUESTED** [review]
@nh2 , great thanks for this fix. I suggested some minor changes.

Also, in next PR, we can do some small code cleanup - splitting lists of packages to separate lines.

**Comment by imincik** [comment]
I tested built binary by importing some test vector data and looks OK (there are some things to fix/improve, but they are not related to this PR)

```
nix-build -A grass
./result/bin/grass
```

screen-imincik-2023-05-31-11:03:54

File: pkgs/applications/gis/grass/default.nix (hunk 2 above)
Comment (nh2) [reply pkgs/applications/gis/grass/default.nix:25]: Done.

File: pkgs/applications/gis/grass/default.nix (hunk 1 above)
Comment (nh2) [reply pkgs/applications/gis/grass/default.nix]: It's a different encoding of the same hash. I use `nix (Nix) 2.3.16` on the machine I did my build on; my machines with newer Nix also generate the other form of hash.

Converting:

```sh
nix --extra-experimental-features nix-command hash to-base64 'sha256:02fxcpdwkjis0i5mqks1by4kwazi1947flmhfd0yz2xxgdvx0wsk'
```

Output:
```
U3PQd3u9i+9Bc7BSd0gK8Ss+iV9BT1xLBDrKydtl3Qk=
```

I've switched it and also use the new `hash` now.

**Review by imincik — APPROVED** [review]
Looks good now. It is good to be merged. Thank you.

**Review by willcohen — APPROVED** [review]

**Pull request merged**